docs/pages/guides/FormikOurValidation.example.tsx

```diff
@@ -0,0 +1,108 @@
+import React from 'react';
+import Code from '../../_shared/Code';
+import { Grid } from '@material-ui/core';
+import { TextField } from '@material-ui/core';
+import { Formik, Form, Field, FieldProps } from 'formik';
+import { format, isWeekend, isWednesday } from 'date-fns';
+import { DatePicker, DatePickerProps } from '@material-ui/pickers';
+
+interface DatePickerFieldProps extends FieldProps, DatePickerProps {
+  getShouldDisableDateError: (date: Date) => string;
+}
+
+const DatePickerField: React.FC<DatePickerFieldProps> = ({
+  form,
+  field: { value, name },
+  maxDate = new Date('2100-01-01'),
+  minDate = new Date('1900-01-01'),
+  getShouldDisableDateError,
+  ...other
+}) => {
+  const currentError = form.errors[name];
+  const toShowError = Boolean(currentError && form.touched[name]);
+
+  return (
+    <DatePicker
+      autoOk
+      clearable
+      minDate={minDate}
+      maxDate={maxDate}
+      value={value}
+      onError={(reason, value) => {
+        switch (reason) {
+          case 'invalidDate':
+            form.setFieldError(name, 'Invalid date format');
+            break;
+
+          case 'disablePast':
+            form.setFieldError(name, 'Values in the past are not allowed');
+            break;
+
+          case 'maxDate':
+            form.setFieldError(name, `Date should not be after ${format(maxDate, 'P')}`);
+            break;
+
+          case 'minDate':
+            form.setFieldError(name, `Date should not be after ${format(maxDate, 'P')}`);
+            break;
+
+          case 'shouldDisableDate':
+            // shouldDisableDate returned true, render custom message according to the `shouldDisableDate` logic
+            form.setFieldError(name, getShouldDisableDateError(value));
+            break;
+
+          default:
+            form.setErrors({
+              ...form.errors,
+              [name]: undefined,
+            });
+        }
+      }}
+      // Make sure that your 3d param is set to `true` on order to not clear errors
+      onChange={date => form.setFieldValue(name, date, false)}
+      renderInput={props => (
+        <TextField
+          name={name}
+          {...props}
+          error={toShowError}
+          helperText={toShowError ? currentError ?? props.helperText : undefined}
+          // Make sure that your 3d param is set to `true` on order to not clear errors
+          onBlur={() => form.setFieldTouched(name, true, false)}
+        />
+      )}
+      {...other}
+    />
+  );
+};
+
+const FormikExample = () => {
```

Comment:
I prefer export default to be always the last export. It is more readable.

Reply:
We systematically use the other way around to 1. increase the code density of the demos, 2. reduce the duplicate of the name of the demo. I think that it's important we still to a single convention.

Reply:
Yeah, but for now pickers are standalone project. And if we are not planning to migrate docs before v5 we could stick to the our internal conventions for now. (before we merge docs)

Reply:
I think that we can break down the dicussion in 3 different topics:

1. **Standaloneness**. The picker components are part of Material-UI, like any other official component. I think that we should treat the component here as they were inside the mono repository. Where the code is a hosted shouldn't influence the developer experience. I think that it should be to the own discretion of how the core team wants to work.
Why? The consistency of the components is one of the important value propositions of the library, I think that steps in this direction improve the DX.
2. **Progressiveness**. I think that we should leverage all the opportunities to reduce the gap in the "standard way". I think that any step in the long term direction is good to take. What the advantage of delaying it? I find it interesting to handle it as we go, as it will be one less change to handle in the future.
3. **Convention evolution**. I think that we should evolve the convention as we learn from our users and the environment change. RFC sounds like the best framework to make changes.
For history, we used to follow the approach of the pull request, it was optimized for reducing the switching cost between class and functional components. With hooks, this requirement is no longer relevant.